### DISCUSSION TITLE
x-data parameters support?

### DISCUSSION THREAD

**Discussion by vladrusu** [alpinejs/alpine#1548] [3. Feature ideas]
I saw that in v3 we cannot submit arguments to the x-data function. Previously, in v2, I used arguments sometimes to create some sort of "dynamic" components, take for example a dropdown with a maxChildren parameter:

```
<div x-data="dropdown(3)">
...
</div>

<div x-data="dropdown(6)">
...
</div>

<script>
function dropdown(maxChildren) {
    return { ... }
}
</script>
```

This is entirely possible in v2. I tried doing this in v3 using the new syntax:

```
<script>
    document.addEventListener('alpine:initializing', () => {
        Alpine.data('dropdown', (maxChildren) => ({
            ...
        }))
    })
</script>
```

But currently, doing this gives this error:

> Uncaught TypeError: Object.defineProperty called on non-object

**Comment by alexej-d** [comment]
This is being discussed [here](https://github.com/alpinejs/alpine/discussions/1521) I believe

**Reply by vladrusu** [reply]
Thanks, didn't know. It can be closed then.
